lib/avo/concerns/has_field_discovery.rb

```diff
@@ -0,0 +1,183 @@
+module Avo
+  module Concerns
+    # This concern facilitates field discovery for models in Avo, mapping database columns and associations to Avo fields.
+    # It supports:
+    #  - Automatic detection of fields based on column names, types, and associations.
+    #  - Customization via `only`, `except`, and global configuration overrides.
+    #  - Handling of special associations like rich text, attachments, and tags.
+    module HasFieldDiscovery
+      extend ActiveSupport::Concern
+
+      DEFAULT_COLUMN_NAMES_MAPPING = {
+        id: { field: "id" },
+        description: { field: "textarea" },
+        gravatar: { field: "gravatar" },
+        email: { field: "text" },
+        password: { field: "password" },
+        password_confirmation: { field: "password" },
+        created_at: { field: "date_time" },
+        updated_at: { field: "date_time" },
+        stage: { field: "select" },
+        budget: { field: "currency" },
+        money: { field: "currency" },
+        country: { field: "country" },
+      }.freeze
+
+      DEFAULT_COLUMN_TYPES_MAPPING = {
+        primary_key: { field: "id" },
+        string: { field: "text" },
+        text: { field: "textarea" },
+        integer: { field: "number" },
+        float: { field: "number" },
+        decimal: { field: "number" },
+        datetime: { field: "date_time" },
+        timestamp: { field: "date_time" },
+        time: { field: "date_time" },
+        date: { field: "date" },
+        binary: { field: "number" },
+        boolean: { field: "boolean" },
+        references: { field: "belongs_to" },
+        json: { field: "code" },
+      }.freeze
+
+      COLUMN_NAMES_TO_IGNORE = %i[
+        encrypted_password reset_password_token reset_password_sent_at remember_created_at password_digest
+      ].freeze
+
+      class_methods do
+        def column_names_mapping
+          @column_names_mapping ||= DEFAULT_COLUMN_NAMES_MAPPING.dup
+                                    .except(*COLUMN_NAMES_TO_IGNORE)
```

Comment:
```suggestion
```

Is this `except` necessary since we're on control of `DEFAULT_COLUMN_NAMES_MAPPING` and those do not include any column from `COLUMN_NAMES_TO_IGNORE`?